lib/src/status.rs

```diff
@@ -324,40 +325,62 @@ pub(crate) async fn status(opts: super::cli::StatusOpts) -> Result<()> {
     Ok(())
 }
 
+/// Write the data for a container image based status.
+fn human_render_imagestatus(
+    mut out: impl Write,
+    slot_name: &str,
+    image: &crate::spec::ImageStatus,
+) -> Result<()> {
+    let transport = &image.image.transport;
+    let imagename = &image.image.image;
+    // Registry is the default, so don't show that
+    let imageref = if transport == "registry" {
+        Cow::Borrowed(imagename)
+    } else {
+        // But for non-registry we include the transport
+        Cow::Owned(format!("{transport}:{imagename}"))
```

Comment:
I need to read about how Cow works but I like this implementation. It does make a lot more sense.

Reply:
Cow basically abstracts over borrowed/owned data here, it's honestly a pointless micro-optimization but it's not hard to do either.